### ISSUE TITLE
How exactly is the PSNR of a model is calculated? 

### ISSUE THREAD

**Issue by 115072** [graphdeco-inria/gaussian-splatting#1078]
I know that PSNR is calculated from the quality of reconstructed images by comparing them to ground truth images, but how is calculated the PSNR of the whole gaussian splatting model?

Furthermore I did a test where took photos of a room in a hemisphere camera position and constructed a gaussian splatting model from it. My first dataset had 20 photos. After the training the value of the PSNR was 35. I did the same training with 100 photos of the same room and the PSNR was just 26. Shouldn't it be better, because I had more details from the room?

**Comment by jiahanyue-del** [comment]
I also encountered the same problem. Have you solved it？

**Comment by 115072** [comment]
Hi.
I read the original gaussian splatting papers and I found out that, they are splitting the input pictures into training and testing groups and they are doing it with every 8th picture. So for example you have 100 photos and it is splitted to 12 testing photos and 88 training photos. After the training is done they make screenshots from the model where the the testing photos supposed to be and compare the screenshots with the ground truth photos. They do it with every photo and in the end they calculate the average PSNR from all the testing photos.
This is how it works, I think.

**Comment by 115072** [comment]
Hi.

I read the original gaussian splatting papers and I found out that, they
are splitting the input pictures into training and testing groups and they
are doing it with every 8th picture. So for example you have 100 photos and
it is splitted to 12 testing photos and 88 training photos. After the
training is done they make screenshots from the model where the the testing
photos supposed to be and compare the screenshots with the ground truth
photos. They do it with every photo and in the end they calculate the
average PSNR from all the testing photos.

This is how it works, I think.

št 24. 7. 2025 o 15:31 jiahanyue-del ***@***.***> napísal(a):

**Comment by jiahanyue-del** [comment]
> 你好。我阅读了原始的高斯喷溅论文，我发现，他们正在将输入图片分成训练组和测试组，并且每 8 张图片就这样做。例如，您有 100 张照片，它被拆分为 12 张测试照片和 88 张训练照片。训练完成后，他们从测试照片应该所在的模型中制作屏幕截图，并将屏幕截图与地面实况照片进行比较。他们对每张照片都这样做，最后他们从所有测试照片中计算出平均 PSNR。我认为这就是它的工作原理。

You're right. Indeed, I found that the psnr provided in the source code is calculated using the images from train. Maybe this is the main reason. Would the psnr calculated using the images from test be more accurate
